frontend/pages/policies/PolicyPage/screens/QueryEditor.tsx

```diff
@@ -79,6 +82,54 @@ const QueryEditor = ({
   const [backendValidators, setBackendValidators] = useState<{
     [key: string]: string;
   }>({});
+  const [aiAutofillData, setAiAutofillData] = useState<{
+    description: string;
+    resolution: string;
+  } | null>(null);
+  const [aiAutofillErrors, setAiAutofillErrors] = useState<any>({});
+  const [isFetchingAIAutofill, setIsFetchingAiAutofill] = useState({
+    description: false,
+    resolution: false,
+  });
+
+  const onAiAutofill = async (fetching: {
```

Comment:
This works but I think desige-wise I would have just had two handlers. one for `onClickAutofillDescrition` and one for `onClickAutofillResolution`.That way you could handle setting the appropriate state values and wouldn't have to worry about creating a `{description, resolution}` object in the children components.

one would look something like this: 

```
const onClickAutofillDescription = () => {
  if (aiAutofillData) {
     setLastEditedQueryDescription(aiAutofillData.description);
  } else {
     updateAutofillInput("description")
  }
  setIsFetchingDescriptionAutofill(false);
}
```

if this changes things too much, I wouldn't worry about changing it.

Reply:
Sweet, okay, next commit will separate the onClick handlers and the fetching states